src/main/java/io/github/opencubicchunks/cubicchunks/core/server/PlayerCubeMap.java

```diff
@@ -557,6 +556,10 @@ public void addPlayer(EntityPlayerMP player) {
         PlayerWrapper playerWrapper = new PlayerWrapper(player);
         playerWrapper.updateManagedPos();
 
+        if (!vanillaNetworkHandler.hasCubicChunks(player)) {
+            vanillaNetworkHandler.updatePlayerPosition(this, player, playerWrapper.getManagedCubePos(), -1);
+        }
+
```

Comment:
Why move this code from below?

Reply:
This call to `updatePlayerPosition` is the first time that a vanilla player will be added to `VanillaNetworkHandler`, and was previously running *after* the player was added to the column/cube watchers. When a player is added to a `ColumnWatcher` that column is sent to the player *immediately* (not on the next tick), meaning that the initial call to `updatePlayerPosition` will resend all the columns (this time in their offset positions) a second time for no reason.

Reply:
Then I think a better way to fix it would be to just have a separate method for adding players in VanillaNetworkhandler, that doesn't actually send anything, and just lets PlayerCubeMap figure it out.